server/service/service_osquery.go

```diff
@@ -153,12 +156,30 @@ func (svc Service) EnrollAgent(ctx context.Context, enrollSecret, hostIdentifier
 		save = true
 	}
 	if save {
-		if err := svc.ds.SaveHost(ctx, host); err != nil {
-			return "", osqueryError{message: "saving host details: " + err.Error(), nodeInvalid: true}
+		if appConfig.ServerSettings.DeferredSaveHost {
+			go svc.serialSaveHost(host)
```

Comment:
Do callers of `EnrollAgent` expect the `host` table to be updated after `EnrollAgent` returns?

Reply:
I believe it's fine, because we have a minor lag anyway between writer and readers, and the accumulation in this serialization mechanism is minimal from what I can see.

@zwass what do you think?